PhpUnit/DefinitionDecoratesConstraint.php

```diff
@@ -0,0 +1,157 @@
+<?php
+
+namespace Matthias\SymfonyDependencyInjectionTest\PhpUnit;
+
+use PHPUnit\Framework\Constraint\Constraint;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+final class DefinitionDecoratesConstraint extends Constraint
+{
+    private const INVALID_BEHAVIORS = [
+        ContainerInterface::RUNTIME_EXCEPTION_ON_INVALID_REFERENCE => 'RUNTIME_EXCEPTION_ON_INVALID_REFERENCE',
+        ContainerInterface::EXCEPTION_ON_INVALID_REFERENCE         => 'EXCEPTION_ON_INVALID_REFERENCE',
+        ContainerInterface::NULL_ON_INVALID_REFERENCE              => 'NULL_ON_INVALID_REFERENCE',
+        ContainerInterface::IGNORE_ON_INVALID_REFERENCE            => 'IGNORE_ON_INVALID_REFERENCE',
+        ContainerInterface::IGNORE_ON_UNINITIALIZED_REFERENCE      => 'IGNORE_ON_UNINITIALIZED_REFERENCE',
+    ];
+
+    private $serviceId;
+    private $decoratedServiceId;
+    private $renamedId;
+    private $priority;
+    private $invalidBehavior;
+
+    public function __construct(string $serviceId, string $decoratedServiceId, ?string $renamedId = null, int $priority = 0, ?int $invalidBehavior = null)
+    {
+        $this->serviceId = $serviceId;
+        $this->decoratedServiceId = $decoratedServiceId;
+        $this->renamedId = $renamedId;
+        $this->priority = $priority;
+        $this->invalidBehavior = $invalidBehavior;
+    }
+
+    public function toString(): string
+    {
+        return sprintf(
+            '"%s" decorates service "%s"%s with priority "%d" and "%s" behavior.',
+            $this->serviceId,
+            $this->decoratedServiceId,
+            $this->renamedId !== null ? sprintf(' and renames it to "%s"', $this->renamedId) : '',
+            $this->priority,
+            self::INVALID_BEHAVIORS[$this->invalidBehavior ?? 0]
+        );
+    }
+
+    public function evaluate($other, string $description = '', bool $returnResult = false): bool
+    {
+        if (!($other instanceof ContainerBuilder)) {
+            throw new \InvalidArgumentException(
+                'Expected an instance of Symfony\Component\DependencyInjection\ContainerBuilder'
+            );
+        }
+
+        if (!$this->evaluateServiceDefinition($other, $returnResult)) {
+            return false;
+        }
+
+        return true;
```

Comment:
This can be replaced with: 

```suggestion
        return $this->evaluateServiceDefinition($other, $returnResult);
```